### PR TITLE
[4.0] Handle PDO driver database creation

### PR DESCRIPTION
**Pull request by roland-d**
Pull Request for Issue #25885 

### Summary of Changes
This change makes sure a database will be created during the installation of Joomla.

### Testing Instructions
1. Get a clean copy of Joomla 4
2. Start the installation process
3. Go through the process until you get to the database part
4. Select here the PDO driver
5. Fill-in a database name that doesn't exist yet
6. Complete the installation
7. Installation doesn't complete because the database name does not exist.
8. Clean the session cookies
9. Apply the patch
10. Restart the installation process
11. Go through the process until you get to the database part
12. Select here the PDO driver
13. Fill-in a database name that doesn't exist yet
14. Complete the installation
15. Installation completes successfully

### Expected result
The database will be created by the installer

### Actual result
The database is not created and throws an error

### Documentation Changes Required
None

### CODE REVIEW

**Comment by brianteeman** [comment]
Many people will actually consider it a  _problem_ that joomla can create a new database in the installation and/or you should not be using a database user who has creation rights. But I guess consistency across all databases is good

File: installation/src/Model/SetupModel.php [hunk 1]
```diff
@@ -354,6 +354,14 @@ public function validateDbConnection()
 			}
 		}
 
+		// Check for PDO drivers
+		$pdo = false;
```
Comment (SharkyKZ) [inline installation/src/Model/SetupModel.php]: Want to simplify like `$pdo = \in_array($options->db_type, ['mysql', 'pgsql']);`?

Comment (roland-d) [reply installation/src/Model/SetupModel.php]: Done.

**Comment by SharkyKZ** [comment]
Works fine on MySQL PDO but getting authentication error on PostgreSQL:

>Could not connect to PDO: SQLSTATE[08006] [7] FATAL: password authentication failed for user

For what it's worth, creating DB on PgSQL doesn't work in 3.x either. So maybe limit this change to MySQL PDO only to restore 3.x behavior?

**Comment by SharkyKZ** [comment]
@brianteeman I have no opinion regarding this. Are you suggesting to remove the ability to create database during installation?

**Comment by brianteeman** [comment]
basically yes. although its useful when building a site locally its something that is pretty unique to joomla iirc

**Comment by roland-d** [comment]
It is not only useful for building sites locally. If you create sites on 
the fly, it is also a very useful feature. Like I do for the pbf testing 
sites for example. I really don't want to create a 100 databases by hand.

Why even remove a useful feature?

**Comment by Eriiond** [comment]
I have tested this item :white_check_mark: successfully on 51fab6a6e8a85051b7e41731710a98c651c23ae2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26273">issues.joomla.org/tracker/joomla-cms/26273</a>.</sub>

**Comment by richard67** [comment]
@roland-d Oh I just see this PR here. I somehow missed it and made an own one, independently from yours, and that has been merged meanwhile: #28338 . So sorry I made your PR here obsolete.

**Comment by richard67** [comment]
Closing as meanwhile solved with #28338 ... again sorry Roland.

**Pull request closed without merging**